src/charm.py

```diff
@@ -740,6 +721,144 @@ def _gen_wp_config(self):
         )
         return "\n".join(wp_config)
 
+    def _container(self):
+        return self.unit.get_container("wordpress")
+
+    def _wordpress_service_exists(self):
+        return "wordpress" in self._container().get_plan().services
+
+    def _stop_server(self):
+        """Stop WordPress (apache) server
+
+        This operation is idempotence.
+        """
+        if self._wordpress_service_exists():
+            self._container().stop("wordpress")
+
+    def _wp_is_installed(self):
+        """Check if WordPress is installed (check if WordPress related tables exist in database)"""
+        process = self._container().exec(["wp", "core", "is-installed"])
+        try:
+            process.wait()
```

Comment:
Do you think a timeout would be beneficial here? Not sure if it is possible/ already enabled by default

Reply:
Added, thanks!